mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/EventDispatcher.java

```diff
@@ -60,13 +63,19 @@
         this.basicProperties = new AMQP.BasicProperties.Builder()
             .headers(ImmutableMap.of(EVENT_BUS_ID, eventBusId.asString()))
             .build();
+
+        this.flux = Flux.create(sink -> this.fluxSink = sink);
```

Comment:
This do not look like something *conventional*...

If you can find something doing this in the official documentation, we may buy it, however, as long as we do not have a *non-hacky* solution, I prefer betting on reactor-rabbitMQ next week release...

Reply:
Well I don't really get everything yet unfortunately, but the other day I saw something about Flux sink in the reactor doc : https://projectreactor.io/docs/core/release/reference/#producing.create

If it can help (or not^^)

Reply:
I think the doc doesn't say you are good to use `sink` outside of `Flux.create()` :) what it did is a trick to pass `sink` to outside then I can use it with no limitation. 

Creating an immediate component to communicate between sink and outside (E.g: A list, queue...) is what I though about. But It takes time to develop and satisfy the contract test. Anw, I'll close this PR as it is not acceptable to use in production